src/App.jsx

```diff
@@ -1,36 +1,85 @@
 import { useEffect, useState } from "react";
 import RestaurantButton from "./Components/RestaurantButton/index.jsx";
 import Hero from "./Components/Hero/index.jsx";
+import MenuItems from "./Components/MenuItems/index.jsx";
+import ReturnButton from "./Components/ReturnButton/index.jsx";
 
 function App() {
-  const [restaurantNames, setRestaurantNames] = useState([]);
-
+  const [restaurantInfo, setRestaurantInfo] = useState([]);
+  const [currentId, setCurrentId] = useState(0);
+  const [currentName, setCurrentName] = useState("");
+  
+  let xlMediaColumn;
+  if (currentId) { xlMediaColumn = "xl:grid-cols-6"};
 
   useEffect(() => {
-    fetch("https://food-delivery-api.dev.io-academy.uk/restaurants")
-      .then((response) => response.json())
-      .then((data) => {
-        setRestaurantNames(data);
-      });
-  }, []);
+    if (!currentId) {
+      fetch("https://food-delivery-api.dev.io-academy.uk/restaurants")
+        .then((response) => response.json())
+        .then((data) => {
+          setRestaurantInfo(data);
+        });
+    } else {
+      fetch(
+        `https://food-delivery-api.dev.io-academy.uk/restaurants/${currentId}`
+      )
+        .then((response) => response.json())
+        .then((data) => {
+          setRestaurantInfo(data.foodItems);
+          setCurrentName(data.restaurant);
+        });
+    }
+  }, [currentId]);
+
+  function renderContent() {
+    if (!currentId) {
+      return restaurantInfo.map((restaurant) => (
+        <RestaurantButton
+        key={restaurant.id}
+        restaurantName={restaurant.name}
+        id={restaurant.id}
+        clickHandler={setCurrentId}
+        />
+      ));
+    } else {
+      return restaurantInfo.map((foodItem, index) => (
+        <MenuItems
+        key={index}
+        foodName={foodItem.foodName}
+        foodType={foodItem.foodType}
+        calories={foodItem.calories}
+        side={foodItem.sideItem}
+        price={Number(foodItem.price).toFixed(2)}
```

Comment:
There are 5 props here that all come from the `foodItem`. Why not just pass the whole `foodItem` object in, and let the component 'unpack' it?

I don't think that the price calculation should happen here. That is definitely the component's job. The `App` should delegate that calculation to this component.